### ISSUE TITLE
Execution failed for task ':app:checkDebugDuplicateClasses'. > Could not resolve all files for configuration ':app:debugRuntimeClasspath'. > Failed to transform jackson-core-2.16.1.jar

### ISSUE THREAD

**Issue by abhijitCN** [stripe/stripe-terminal-react-native#653]
unable to run @stripe/stripe-terminal-react-native in latest react-native 0.73.6
FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
Failed to transform jackson-core-2.16.1.jar (com.fasterxml.jackson.core:jackson-core:2.16.1) to match attributes {artifactType=enumerated-runtime-classes, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime}.

I am using =>
{
buildToolsVersion = "34.0.0"
minSdkVersion = 26
compileSdkVersion = 34
targetSdkVersion = 34
ndkVersion = "25.1.8937393"
kotlinVersion = "1.8.0"
}
and
"dependencies": {
"@stripe/stripe-terminal-react-native": "^0.0.1-beta.16",
"react": "18.2.0",
"react-native": "0.73.6"
}
and gradle-8.3-all.zip for Android which version is wrong for me
![Screenshot 2024-03-29 at 2 05 00 PM](https://github.com/stripe/stripe-terminal-react-native/assets/113179261/98304710-7c7e-4584-9f32-b7e9235b72c4)

**Comment by nazli-stripe** [comment]
it looks like your project's minSDK is set at 21, but we require at least 26: https://github.com/stripe/stripe-terminal-react-native?tab=readme-ov-file#android

**Comment by abhijitCN** [comment]
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.    
   > Failed to transform jackson-core-2.16.1.jar (com.fasterxml.jackson.core:jackson-core:2.16.1) to match attributes {artifactType=enumerated-runtime-classes, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime}.    
      > Execution failed for JetifyTransform: C:\Users\FN\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-core\2.16.1\9456bb3cdd0f79f91a5f730a1b1bb041a380c91f\jackson-core-2.16.1.jar.
         > Failed to transform 'C:\Users\FN\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-core\2.16.1\9456bb3cdd0f79f91a5f730a1b1bb041a380c91f\jackson-core-2.16.1.jar' using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 65. (Run with --stacktrace for more 
details.)
![Capture](https://github.com/stripe/stripe-terminal-react-native/assets/113179261/2eb63192-160b-4f28-a020-4a63eff61b85)

now I use =>
I am using =>
{
buildToolsVersion = "34.0.0"
minSdkVersion = 26
compileSdkVersion = 34
targetSdkVersion = 34
ndkVersion = "25.1.8937393"
kotlinVersion = "1.8.0"
}
and
"dependencies": {
"@stripe/stripe-terminal-react-native": "^0.0.1-beta.16",
"react": "18.2.0",
"react-native": "0.73.6"
}
and gradle-8.3-all.zip for Android also using java version "17.0.10" 2024-01-16 LTS

> it looks like your project's minSDK is set at 21, but we require at least 26: https://github.com/stripe/stripe-terminal-react-native?tab=readme-ov-file#android

![Capture](https://github.com/stripe/stripe-terminal-react-native/assets/113179261/da60f776-f2e0-44e2-9027-ff18bb4b0c0d)

**Comment by DustinBracy** [comment]
I'm having the same issue, not sure how to get around it.

**Comment by abhijitCN** [comment]
> I'm having the same issue, not sure how to get around it.

Hello, can you please share all the versions along with Java, Kotln, React-native, minSdkVersion, targetSdkVersion, and compileSdkVersion that version? and in iOS your project is running fine? please share

**Comment by DustinBracy** [comment]
> > I'm having the same issue, not sure how to get around it.
> 
> Hello, can you please share all the versions along with Java, Kotln, React-native, minSdkVersion, targetSdkVersion, and compileSdkVersion that version? and in iOS your project is running fine? please share

My iOS app builds fine. Android fails with the issues you mentioned. I'm using target version 33, compile version 34, min version 26, react 18.2, react Native 73.6. I can build fine using the previous version and expo 49  (0.0.1-beta.15), but I'd like to use expo 50 so I can better integrate sentry

**Comment by Zaynmiraj** [comment]
> unable to run @stripe/stripe-terminal-react-native in latest react-native 0.73.6 FAILURE: Build failed with an exception.
> 
> What went wrong: Execution failed for task ':app:checkDebugDuplicateClasses'. Could not resolve all files for configuration ':app:debugRuntimeClasspath'. Failed to transform jackson-core-2.16.1.jar (com.fasterxml.jackson.core:jackson-core:2.16.1) to match attributes {artifactType=enumerated-runtime-classes, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime}.
> 
> I am using => { buildToolsVersion = "34.0.0" minSdkVersion = 26 compileSdkVersion = 34 targetSdkVersion = 34 ndkVersion = "25.1.8937393" kotlinVersion = "1.8.0" } and "dependencies": { "@stripe/stripe-terminal-react-native": "^0.0.1-beta.16", "react": "18.2.0", "react-native": "0.73.6" } and gradle-8.3-all.zip for Android which version is wrong for me ![Screenshot 2024-03-29 at 2 05 00 PM](https://private-user-images.githubusercontent.com/113179261/317998716-98304710-7c7e-4584-9f32-b7e9235b72c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMTgzNzIsIm5iZiI6MTcxMjAxODA3MiwicGF0aCI6Ii8xMTMxNzkyNjEvMzE3OTk4NzE2LTk4MzA0NzEwLTdjN2UtNDU4NC05ZjMyLWI3ZTkyMzViNzJjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMlQwMDM0MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNDE2NDQyNzg0N2MxMTUxZmQ4OGQ0MDhhYjMxMmQyMzc0YmQwYzNmOTJjYzU4Y2VlZjNhOWI0MDRiOTY0ODlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.DCPos_xgHO4Sf0rPrGYZX7R8n8Z27RXfmaM9bAF-M8k)

Same problem I am getting

**Comment by Zaynmiraj** [comment]
> > > I'm having the same issue, not sure how to get around it.
> > 
> > 
> > Hello, can you please share all the versions along with Java, Kotln, React-native, minSdkVersion, targetSdkVersion, and compileSdkVersion that version? and in iOS your project is running fine? please share
> 
> My iOS app builds fine. Android fails with the issues you mentioned. I'm using target version 33, compile version 34, min version 26, react 18.2, react Native 73.6. I can build fine using the previous version and expo 49 (0.0.1-beta.15), but I'd like to use expo 50 so I can better integrate sentry

how to use in expo ?

**Comment by nazli-stripe** [comment]
this is an issue due to us underlying the underlying Terminal Android SDK to `3.4.0`: https://github.com/stripe/stripe-terminal-android/issues/433#issuecomment-1970088784

If you update your `gradle.properties` to include either
`android.jetifier.ignorelist = jackson-core`
or 
`android.enableJetifier = false`

does it resolve the issue?

**Comment by DustinBracy** [comment]
> this is an issue due to us underlying the underlying Terminal Android SDK to `3.4.0`: [stripe/stripe-terminal-android#433 (comment)](https://github.com/stripe/stripe-terminal-android/issues/433#issuecomment-1970088784)
> 
> If you update your `gradle.properties` to include either `android.jetifier.ignorelist = jackson-core` or `android.enableJetifier = false`
> 
> does it resolve the issue?

Still does not resolve the issue.  I tried removing the `android.enableJetifier=true` line completely, changing it to `android.enableJetifier=false`, and tried `android.jetifier.ignorelist=jackson-core-2.16.1.jar` and `android.jetifier.ignorelist=jackson-core` underneath `android.enableJetifier=true` in gradle.properties after running npx expo prebuild to generate the android directory.  Still getting the same error.

**Comment by abhijitCN** [comment]
> ignorelist



> this is an issue due to us underlying the underlying Terminal Android SDK to `3.4.0`: [stripe/stripe-terminal-android#433 (comment)](https://github.com/stripe/stripe-terminal-android/issues/433#issuecomment-1970088784)
> 
> If you update your `gradle.properties` to include either `android.jetifier.ignorelist = jackson-core` or `android.enableJetifier = false`
> 
> does it resolve the issue?



> this is an issue due to us underlying the underlying Terminal Android SDK to `3.4.0`: [stripe/stripe-terminal-android#433 (comment)](https://github.com/stripe/stripe-terminal-android/issues/433#issuecomment-1970088784)
> 
> If you update your `gradle.properties` to include either `android.jetifier.ignorelist = jackson-core` or `android.enableJetifier = false`
> 
> does it resolve the issue?



> this is an issue due to us underlying the underlying Terminal Android SDK to `3.4.0`: [stripe/stripe-terminal-android#433 (comment)](https://github.com/stripe/stripe-terminal-android/issues/433#issuecomment-1970088784)
> 
> If you update your `gradle.properties` to include either `android.jetifier.ignorelist = jackson-core` or `android.enableJetifier = false`
> 
> does it resolve the issue?

thank you @nazli-stripe app built successfully using android.enableJetifier = false in latest teact native project 
USING =>
        "react": "18.2.0",
        "react-native": "0.73.6",
        buildToolsVersion = "34.0.0"
        minSdkVersion = 26
        compileSdkVersion = 34
        targetSdkVersion = 33
        ndkVersion = "25.1.8937393"
        kotlinVersion = "1.8.0"

**Comment by Eonfuzz** [comment]
For those on expo you can write a plugin to edit the gradle.properties file

```javascript
const { withGradleProperties } = require('@expo/config-plugins');

/**
 * tl;dr fix for
 * https://github.com/stripe/stripe-terminal-react-native/issues/653
 *
 * This is a workaround for an issue with the Stripe Terminal SDK and Jetifier.
 * Should be able to remove this once stripe is updated to sdk 18
 * @param {*} config
 * @param {*} customName
 * @returns
 */

const withJetifierIgnoringJackson = (config, customName) => {
  return withGradleProperties(config, async config => {
    config.modResults.push({
      key: 'android.jetifier.ignorelist',
      value: 'jackson-core',
      type: 'property',
    });

    return config;
  });
};
/// Use the plugin
module.exports = withJetifierIgnoringJackson;

```

**Comment by Brunowilliang** [comment]
> For those on expo you can write a plugin to edit the gradle.properties file
> 
> yourplugin.js
> 
> ```js
> const { withGradleProperties } = require('@expo/config-plugins');
> 
> /**
>  * tl;dr fix for
>  * https://github.com/stripe/stripe-terminal-react-native/issues/653
>  *
>  * This is a workaround for an issue with the Stripe Terminal SDK and Jetifier.
>  * Should be able to remove this once stripe is updated to sdk 18
>  * @param {*} config
>  * @param {*} customName
>  * @returns
>  */
> 
> const withJetifierIgnoringJackson = (config, customName) => {
>   return withGradleProperties(config, async config => {
>     config.modResults.push({
>       key: 'android.jetifier.ignorelist',
>       value: 'jackson-core',
>       type: 'property',
>     });
> 
>     return config;
>   });
> };
> /// Use the plugin
> module.exports = withJetifierIgnoringJackson;
> ```
> 
> and then in appconfig
> 
> ```
> {
>     ...
>     plugins: ['./pathto/yourplugin']
> }
> ```

Damn, after about 2 days of solving this bug, this solution worked perfectly for me, thank you

**Comment by DustinBracy** [comment]
> For those on expo you can write a plugin to edit the gradle.properties file
> 
> yourplugin.js
> 
> ```js
> const { withGradleProperties } = require('@expo/config-plugins');
> 
> /**
>  * tl;dr fix for
>  * https://github.com/stripe/stripe-terminal-react-native/issues/653
>  *
>  * This is a workaround for an issue with the Stripe Terminal SDK and Jetifier.
>  * Should be able to remove this once stripe is updated to sdk 18
>  * @param {*} config
>  * @param {*} customName
>  * @returns
>  */
> 
> const withJetifierIgnoringJackson = (config, customName) => {
>   return withGradleProperties(config, async config => {
>     config.modResults.push({
>       key: 'android.jetifier.ignorelist',
>       value: 'jackson-core',
>       type: 'property',
>     });
> 
>     return config;
>   });
> };
> /// Use the plugin
> module.exports = withJetifierIgnoringJackson;
> ```
> 
> and then in appconfig
> 
> ```
> {
>     ...
>     plugins: ['./pathto/yourplugin']
> }
> ```

This works for me as well, thank you @Eonfuzz!!

**Comment by nazli-stripe** [comment]
closing this one as there is a working fix, thanks @Eonfuzz for the patch!

**Comment by isaachinman** [comment]
@nazli-stripe When will you update to sdk 18?

**Comment by nazli-stripe** [comment]
@isaachinman next release, should be by end of june

**Comment by Zaynmiraj** [comment]
android/gradle.properties 
Change it 
android.enableJetifier=true
To 
android.enableJetifier=false

**Comment by isaachinman** [comment]
@Eonfuzz Wouldn't it be possible to achieve that without a custom plugin, via `expo-build-properties`?

**Comment by hungcaovn** [comment]
> For those on expo you can write a plugin to edit the gradle.properties file
> 
> yourplugin.js
> 
> ```js
> const { withGradleProperties } = require('@expo/config-plugins');
> 
> /**
>  * tl;dr fix for
>  * https://github.com/stripe/stripe-terminal-react-native/issues/653
>  *
>  * This is a workaround for an issue with the Stripe Terminal SDK and Jetifier.
>  * Should be able to remove this once stripe is updated to sdk 18
>  * @param {*} config
>  * @param {*} customName
>  * @returns
>  */
> 
> const withJetifierIgnoringJackson = (config, customName) => {
>   return withGradleProperties(config, async config => {
>     config.modResults.push({
>       key: 'android.jetifier.ignorelist',
>       value: 'jackson-core',
>       type: 'property',
>     });
> 
>     return config;
>   });
> };
> /// Use the plugin
> module.exports = withJetifierIgnoringJackson;
> ```
> 
> and then in appconfig
> 
> ```
> {
>     ...
>     plugins: ['./pathto/yourplugin']
> }
> ```

@Eonfuzz : I am getting this error with "require('@expo/config-plugins');". Did you get this? 
<img width="309" alt="Screenshot 2024-04-07 at 10 02 43" src="https://github.com/stripe/stripe-terminal-react-native/assets/13977238/858b272c-4828-451a-b2fa-3de6ad45ac51">

**Comment by Eonfuzz** [comment]
> @Eonfuzz Wouldn't it be possible to achieve that without a custom plugin, via `expo-build-properties`?

I couldn't see how to edit the specific gradle prop I was after in that documentation, if you find out a way please let me know!

@hungcaovn do you have @expo/config-plugins installed?
https://www.npmjs.com/package/@expo/config-plugins

**Comment by hungcaovn** [comment]
@Eonfuzz : Yes I did

**Comment by a-eid** [comment]
@nazli-stripe I don't think this is an acceptable solution.

**Comment by mykel-unwritten** [comment]
changing `android.jettifier` to `false` produces a different issue mentioned here
https://medium.com/@hasperong/duplicate-class-android-support-v4-app-inotificationsidechannel-found-in-modules-classes-1719262dfdcd

and the fix is to enable jettifier 😓

**Comment by mykel-unwritten** [comment]
disabling jettifier shows me this error, could this be an issue from another library?
<img width="1440" alt="Screenshot 2024-04-29 at 8 58 16 PM" src="https://github.com/stripe/stripe-terminal-react-native/assets/167043487/855972d1-1e70-4d70-bac3-b24465e9bd52">

**Comment by Eonfuzz** [comment]
@nazli-stripe  given this the "fix" for expo to work correctly with stripe-terminal-react-native could you please add it to your expo setup file? 

https://github.com/stripe/stripe-terminal-react-native/blob/afe3fe89483a29e6e91e1e03547f519c94f346e0/src/plugin/withStripeTerminal.ts#L32

**Comment by nazli-stripe** [comment]
@Eonfuzz added in https://github.com/stripe/stripe-terminal-react-native/pull/754

**Comment by nazli-stripe** [comment]
beta.20 release updated expo config to automatically disable jettifier for jackson-core

**Comment by DenisDov** [comment]
in your file `gradle.properties` add this, below `android.enableJetifier=true`:

`android.jetifier.ignorelist=jackson-core`
